avro-converter/src/test/java/io/confluent/copycat/avro/AvroDataTest.java

```diff
@@ -0,0 +1,671 @@
+                 AvroData.toCopycatData(avroSchema, avroRecord));
+  }
+
+  @Test
+  public void testToCopycatSchemaless() {
+    GenericRecord avroNullRecord = new GenericRecordBuilder(AvroData.ANYTHING_SCHEMA).build();
+    assertEquals(new SchemaAndValue(null, null),
+                 AvroData.toCopycatData(AvroData.ANYTHING_SCHEMA, avroNullRecord));
+
+    GenericRecord avroIntRecord = new GenericRecordBuilder(AvroData.ANYTHING_SCHEMA)
+        .set("int", 12)
+        .build();
+    assertEquals(new SchemaAndValue(null, 12),
+                 AvroData.toCopycatData(AvroData.ANYTHING_SCHEMA, avroIntRecord));
+
+    GenericRecord avroLongRecord = new GenericRecordBuilder(AvroData.ANYTHING_SCHEMA)
+        .set("long", 12L)
+        .build();
+    assertEquals(new SchemaAndValue(null, 12L),
+                 AvroData.toCopycatData(AvroData.ANYTHING_SCHEMA, avroLongRecord));
+
+    GenericRecord avroFloatRecord = new GenericRecordBuilder(AvroData.ANYTHING_SCHEMA)
+        .set("float", 12.2f)
+        .build();
+    assertEquals(new SchemaAndValue(null, 12.2f),
+                 AvroData.toCopycatData(AvroData.ANYTHING_SCHEMA, avroFloatRecord));
+
+    GenericRecord avroDoubleRecord = new GenericRecordBuilder(AvroData.ANYTHING_SCHEMA)
+        .set("double", 12.2)
+        .build();
+    assertEquals(new SchemaAndValue(null, 12.2),
+                 AvroData.toCopycatData(AvroData.ANYTHING_SCHEMA, avroDoubleRecord));
+
+    GenericRecord avroBooleanRecord = new GenericRecordBuilder(AvroData.ANYTHING_SCHEMA)
+        .set("boolean", true)
+        .build();
+    assertEquals(new SchemaAndValue(null, true),
+                 AvroData.toCopycatData(AvroData.ANYTHING_SCHEMA, avroBooleanRecord));
+
+    GenericRecord avroStringRecord = new GenericRecordBuilder(AvroData.ANYTHING_SCHEMA)
+        .set("string", "teststring")
+        .build();
+    assertEquals(new SchemaAndValue(null, "teststring"),
+                 AvroData.toCopycatData(AvroData.ANYTHING_SCHEMA, avroStringRecord));
+
+
+    GenericRecord avroArrayRecord = new GenericRecordBuilder(AvroData.ANYTHING_SCHEMA)
+        .set("array", Arrays.asList(avroIntRecord, avroStringRecord))
+        .build();
+    assertEquals(new SchemaAndValue(null, Arrays.asList(12, "teststring")),
+                 AvroData.toCopycatData(AvroData.ANYTHING_SCHEMA, avroArrayRecord));
+
+    GenericRecord avroMapEntry = new GenericRecordBuilder(AvroData.ANYTHING_SCHEMA_MAP_ELEMENT)
+        .set("key", avroIntRecord)
+        .set("value", avroStringRecord)
+        .build();
+    GenericRecord avroMapRecord = new GenericRecordBuilder(AvroData.ANYTHING_SCHEMA)
+        .set("map", Arrays.asList(avroMapEntry))
+        .build();
+    assertEquals(new SchemaAndValue(null, Collections.singletonMap(12, "teststring")),
+                 AvroData.toCopycatData(AvroData.ANYTHING_SCHEMA, avroMapRecord));
+
+  }
+
+  private static NonRecordContainer assertNonRecordConversion(
```

Comment:
Only because the assert does the conversion and in one case I want to do one additional check. I've changed the name to `checkNonRecordConversion` so it is not misleading.